### PR TITLE
[2.11.x] DDF-3385 Apply pklinef fix for filter injector servlet context race condition.

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Apply pklinef fix for filter injector servlet context race condition.
#### Who is reviewing it? 
@pklinef @shaundmorris 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@lessarderic 
@rzwiefel 
#### How should this be tested? (List steps with links to updated documentation)
Install and verify there are no errors related to injecting the DelegateServletFilter
#### What are the relevant tickets?
[DDF-3385](https://codice.atlassian.net/browse/DDF-3385)

### CODE REVIEW

**Comment by clockard** [comment]
build now

File: platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java [hunk 1]
```diff
@@ -61,18 +66,25 @@ public FilterInjector(Filter filter) {
     this.delegateServletFilter = filter;
   }
 
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
```
Comment (rzwiefel) [inline platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/FilterInjector.java]: ❓ Is there any need to nullcheck the bundle or bundleContext first?

File: platform/platform-filter-delegate/src/test/java/org/codice/ddf/platform/filter/delegate/FilterInjectorTest.java [hunk 2]
```diff
@@ -50,13 +53,14 @@ public void testInjectFilter() {
     FilterInjector injector = new FilterInjector(filter);
     updateMockReference();
 
-    injector.injectFilter(curReference);
+    injector.event(curEvent, null);
 
     verify(curContext).addFilter("delegating-filter", filter);
   }
```
Comment (lessarderic) [inline platform/platform-filter-delegate/src/test/java/org/codice/ddf/platform/filter/delegate/FilterInjectorTest.java]: ❗️ We should add these unit tests:

```
  @Test
  public void testInjectFilterHandlesOnlyServletContext() {
    curEvent = mock(ServiceEvent.class);
    curReference = mock(ServiceReference.class);
    curContext = mock(Servlet.class);
    Bundle bundle = mock(Bundle.class);
    BundleContext context = mock(BundleContext.class);

    when(curEvent.getType()).thenReturn(ServiceEvent.REGISTERED);
    when(curEvent.getServiceReference()).thenReturn(curReference);
    when(curReference.getBundle()).thenReturn(bundle);
    when(bundle.getBundleContext()).thenReturn(context);
    when(context.getService(curReference)).thenReturn(curContext);

    injector.event(curEvent, null);

    verify(curContext, never()).addFilter("delegating-filter", filter);
  }

  @Test
  public void testInjectFilterIgnoresUnregisteringEvents() {
    curEvent = mock(ServiceEvent.class);
    when(curEvent.getType()).thenReturn(ServiceEvent.UNREGISTERING);

    injector.event(curEvent, null);

    verify(curEvent, never()).getServiceReference();
  }

  @Test
  public void testInjectFilterIgnoresModifiedEvents() {
    curEvent = mock(ServiceEvent.class);
    when(curEvent.getType()).thenReturn(ServiceEvent.MODIFIED);

    injector.event(curEvent, null);

    verify(curEvent, never()).getServiceReference();
  }

  @Test
  public void testInjectFilterIgnoresModifiedEndMatchEvents() {
    curEvent = mock(ServiceEvent.class);
    when(curEvent.getType()).thenReturn(ServiceEvent.MODIFIED_ENDMATCH);

    injector.event(curEvent, null);

    verify(curEvent, never()).getServiceReference();
  }
```

Comment (clockard) [reply platform/platform-filter-delegate/src/test/java/org/codice/ddf/platform/filter/delegate/FilterInjectorTest.java]: Added requested unit tests.

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
Manually verified on my local machine but it wasn't showing the issue previously either.

**Pull request merged**